query/iterator.go

```diff
@@ -815,6 +815,8 @@ func (opt IteratorOptions) Window(t int64) (start, end int64) {
 		start = t - dt
 	}
 
+	start += int64(opt.Interval.Offset)
```

Comment:
Can you add some tests for this behavior?

Reply:
Can do.  @gshif  developed a truly minimal repro case I can encapsulate in a test.